### PR TITLE
Feat: Project tiles on the home page

### LINKED ISSUE #84: Projects tiles on the home page
Add project tiles components on the home page

### PR DESCRIPTION
**Pull request by dkostenko-bjerk**
- Added project tiles box component
- Added images for projects in /static folder
- Added mdx support for projects pages

Close #84 #85 #86 #87 #88

### CODE REVIEW

File: web/src/pages/index.tsx [hunk 1]
```diff
@@ -33,6 +36,29 @@ const actionBoxMockup: CallToActionBoxProps[] = [
...
+    image: 'lent.png',
+    linkText: 'Lent – Læringsapplikasjon',
+    linkTo: '/projects/lent',
+  },
+  {
+    image: 'indiv.png',
+    linkText: 'Indiv',
+    linkTo: '/projects/indiv',
+  },
+];
```
Comment (cobraz) [inline web/src/pages/index.tsx:60]: This should probably be handled with `frontmatter`.

Example:

Filename: `web/src/projects/tabetalt.mdx`
```typescript
---
slug: "/projects/tabetalt"
title: "Tabetalt – En nettbutikkleverandør"
---

...
```

Which make it possible for us to do this:

```suggestion
const projectsBoxesData: CallToActionImageProps[] = useStaticQuery(graphql`
	query($slug: String!) {
    markdownRemark(frontmatter: { slug: { eq: $slug } }) {
      frontmatter {
        slug
        title
      }
    }
`);
```

**Note**: What about internalization? re #23 

### References

- [StaticQuery](https://www.gatsbyjs.com/docs/how-to/querying-data/use-static-query/)
- https://www.gatsbyjs.com/docs/how-to/routing/adding-markdown-pages/

**Review by cobraz — CHANGES_REQUESTED** [review]
Thank you for this!

Could we maybe do something about the `projectsBoxesData` variable: I've left an idea surrounding [frontmatter](https://coffeecodeclimb.com/what-is-frontmatter/). 

**Nitpicking:** We should consider having a dedicated folder for assets related to projects.

**Review by AnneMatilde — APPROVED** [review]
This looks good, design wise.

File: web/src/pages/index.tsx (hunk 1 above)
Comment (dkostenko-bjerk) [reply web/src/pages/index.tsx:66]: @cobraz I added mdx support in two last commits but there were a few questions:

Now there are providing components in .mdx files, I can change them to the md-tags, like here
```jsx
// template.js
import { H1Component, H2Component, H3Component } from './components';
...
<MDXProvider
  components={{
    h1: H1Component,
    h2: H2Component,
    h3: H3Component,
    ...
  }}
>
...
```
```markdown
// project.mdx
---
slug: "/project"
title: "title"
image: "image.jpg"
description: "Description"
---
## title2
### title3
text
...
```

But some of the pages (e.g. project pages) have different styles (margins, etc.).
Which is better: provide components as md-tags or use them directly in .mdx files?

Comment (cobraz) [reply web/src/pages/index.tsx:66]: Its probably a better user experience to provide them as md-tags, right? What do you think?

**Review by AnneMatilde — APPROVED** [review]
I dont see any design changes so design wise, this is approved.

File: web/src/pages/index.tsx (hunk 1 above)
Comment (dkostenko-bjerk) [reply web/src/pages/index.tsx:66]: @cobraz  I think yes, but how to style, for example, such symbols/images with md-tags without creating additional styles and components or to create flexible images?
image

Comment (cobraz) [reply web/src/pages/index.tsx:66]: We can have props for md-tags, right?

Comment (dkostenko-bjerk) [reply web/src/pages/index.tsx:66]: @cobraz yes, but if we create an mdx template for the project model, these props will be the same on each of the pages, right?
```jsx
// templates/project.js
const Heading = ({ children, ...props }) => (
   <BaseHeading {...props}>{children}</BaseHeading>
 );

 export default function ProjectTemplate({ data: { mdx } }) {
   return (
      ...
           <MDXProvider components={{ h1: Heading }}>
             <MDXRenderer>{mdx.body}</MDXRenderer>
           </MDXProvider>
      ...
   );
 }
```
```markdown
// /pages/project1.mdx
---
slug: "/projects/project1"
title: "Title"
...
---
# Heading component text
```

Comment (cobraz) [reply web/src/pages/index.tsx:66]: Yes, I would probably the `mdx` files in an other location tho'.

e.g. `src/web/content/projects/`?

Comment (cobraz) [reply web/src/pages/index.tsx:66]: btw, you probably have to use [Gatsby Node API](https://www.gatsbyjs.com/docs/reference/config-files/gatsby-node/) for this btw.

**Review by AnneMatilde — APPROVED** [review]
Looking good!

**Comment by dkostenko-bjerk** [comment]
I added markdown support for home page yet (and hardcode localization)

**Review by cobraz — APPROVED** [review]
👏

**Pull request merged**